pysnmp/proto/secmod/eso/priv/des3.py

```diff
@@ -51,17 +52,19 @@ def hashPassphrase(self, authProtocol, privKey):
                 'Unknown auth protocol %s' % (authProtocol,)
             )
 
-    # 2.1
+    #key localization as per https://tools.ietf.org/html/draft-reeder-snmpv3-usm-3desede-00
     def localizeKey(self, authProtocol, privKey, snmpEngineID):
         if authProtocol == hmacmd5.HmacMd5.serviceID:
             localPrivKey = localkey.localizeKeyMD5(privKey, snmpEngineID)
-            for count in range(1, int(ceil(self.keySize * 1.0 / len(localPrivKey)))):
-                # noinspection PyDeprecation,PyCallingNonCallable
-                localPrivKey += localkey.localizeKeyMD5(localPrivKey, snmpEngineID)
+            #now extend this key if too short by repeating steps that includes the hashPassphrase step
+            while (len(localPrivKey) < self.keySize):
+                newKey = hashPassphraseMD5(localPrivKey)
```

Comment:
Could you please elaborate why passphrase hashing (via `hashPassphraseMD5 `) is required here? I can't find that step in [RFC](https://tools.ietf.org/html/draft-reeder-snmpv3-usm-3desede-00):

    First, run the password-to-key algorithm with inputs of the passphrase and engineID as described in
    the USM document. Secondly, run the password-to-key algorithm again with the previous output
    (instead of the passphrase) and the same engineID as inputs.

Or in pseudocode:

    K1 <-- P2K( <passphrase>, <engine_id> )
    K2 <-- P2K( K1,           <engine_id> )
    localized_key = K1 | K2